docs/scos/dev/front-end-development/202307.0/oryx/building-components/component-definition.md

```diff
@@ -0,0 +1,46 @@
+---
+title: "Oryx components definition"
+description: Components are registered in an Oryx application by a definition file
+last_updated: Sept 19, 2023
+template: concept-topic-template
+---
+
+Oryx components can be used in different ways. They can be configured in [pages](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-pages/oryx-pages.html) and [compositions](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-pages/oryx-compositions.html), used in components, or integrated in CMS content.
+
+When a component is rendered for the first time, Oryx resolves the component definition from the registry and loads the associated implementation. With this, components are lazily loaded.
+
+## Creating a component definition
+
+To register a [component implementation](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-components/component-implementation.html), you need to provide a component definition. The component definition requires a name and an implementation. The name is used as the web component element name and consists of two or more words separated by a dash. We recommend prefixing component names with a project, brand, or company name. For example, Oryx components are prefixed with `oryx-`.
+
+{% info_block infoBox "Update definitions" %}
+You can also update an existing component definition. To match an existing definition, you still need to provide a name.
+{% endinfo_block %}
+
+The following example shows where a component is registered, providing both the name and implementation.
```

Comment:
Why are we showing how to _register_ a components when this section is about _creating_ a definition?

Reply:
Fair. I've dropped it, since this part is covered in the building component doc.
The component definition will get more meat over time, but for now we only documented the basics.